src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java

```diff
@@ -97,6 +104,9 @@
             return castOrNull(key,getExecution());
         } else if (FlowNode.class.isAssignableFrom(key)) {
             return castOrNull(key, getNode());
+        } else if (key == EnvironmentExpander.class) {
```

Comment:
Maybe I should move this case up next to the `TaskListener` case?

Reply:
Do you think the password parameter case may not get reached?

Reply:
No, just for consistency/style, since the cases here are checking `isAssignableFrom`, I think in case someone looks up a subclass of the type, but for this case, I don't think that would make sense, and the direct-equality cases are handled above.

Reply:
Moved up in b142593 (I force-pushed over the old commit)